corehq/apps/users/views/__init__.py

```diff
@@ -878,6 +870,17 @@ def delete_user_role(request, domain):
     if not domain_has_privilege(domain, privileges.ROLE_BASED_ACCESS):
         return JsonResponse({})
     role_data = json.loads(request.body.decode('utf-8'))
+    user_count = get_role_user_count(domain, role_data["_id"])
+    if user_count:
+        return JsonResponse({
```

Comment:
I don't think this will get shown to the user, assuming [this](https://github.com/dimagi/commcare-hq/blob/71c5a0cf7ba92a23e7e6562dc1f833a14ade3d15/corehq/apps/users/static/users/js/roles.js#L355) is the deletion handler. That looks easy to address, like [this](https://github.com/dimagi/commcare-hq/blob/71c5a0cf7ba92a23e7e6562dc1f833a14ade3d15/corehq/apps/users/static/users/js/roles.js#L374).

Reply:
I thought the same thing but there is a handler added by the knockout 'deleteButton' binding handler: https://github.com/dimagi/commcare-hq/blob/98d3e8dcf83bd7716f060ee0645cf42f5f87898f/corehq/apps/hqwebapp/static/hqwebapp/js/main.js#L223

Reply:
aha, great